### ISSUE TITLE
Nested Kotlin objects in the default package cannot be accessed from Gradle Groovy scripts

### ISSUE THREAD

**Issue by mtrakal** [gradle/gradle#9251]
`gradleVersion = '4.10'`

root project build.gradle (not kotlin file)
```
buildscript {
    dependencies {
        classpath XCfg.BuildPlugins.android_plugin
    }
}
```

buildSrc\build.gradle.kts
```
repositories {
    jcenter()
}
plugins {
    `kotlin-dsl`
}
```

buildSrc\src\main\java\XCfg.kt
```
object XCfg {
    object Versions {
        val kotlinVersion = "1.2.61"
        val gradleBuildTools = "3.3.0-alpha03"
    }

    object BuildPlugins {
        val android_plugin = "com.android.tools.build:gradle:${Versions.gradleBuildTools}"
        val kotlin_plugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:${Versions.kotlinVersion}"
    }
}
```

### Expected Behavior
Should compile without any error

### Current Behavior
Failed on: ` No such property: BuildPlugins for class: XCfg`

### Steps to Reproduce (for bugs)
See file configuration on top of this issue.

When just change files to this, it start compiling and working.

root project build.gradle (not kotlin file)
```
buildscript {
    dependencies {
        classpath BuildPlugins.android_plugin
    }
}
```

buildSrc\src\main\java\XCfg.kt
```
object BuildPlugins {
    val android_plugin = "com.android.tools.build:gradle:${Versions.gradleBuildTools}"
    val kotlin_plugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:${Versions.kotlinVersion}"
}

object Versions {
    val kotlinVersion = "1.2.61"
    val gradleBuildTools = "3.3.0-alpha03"
}

object XCfg {
}
```

But I think, that inner object should work for kotlin same as for groovy gradle files.
In https://expertise.jetruby.com/how-to-write-gradle-scripts-with-kotlin-dsl-better-than-groovy-a63ffc9be2e3 they use this config, but for gradle kotlin files, not groovy. But I don't want to migrate all groovy files to kotlin files in same time for now...

**Comment by extmkv** [comment]
I've been tracking this issue and after 2y still reproducible

**Comment by arberg** [comment]
Yep, I also got stuck on this issue today, migrating from groovy to kotlin.

**Comment by iamarjun** [comment]
+1

**Comment by CMingTseng** [comment]
i use android studio arctic fox stable version 

i still get the same error  ----    "not in a package"  

@StefMa

**Comment by CMingTseng** [comment]
> Ah ok, haven't noticed the "not in a package" 🙃
> Seems you have the same issue like #807 and #1051.
> According to the [comment there](https://github.com/gradle/kotlin-dsl/issues/1051#issuecomment-415593461) it was fixed 10 days ago.
> The fix is not containing in the 4.10 release but probably in the latest 5.0 nightly.
> 
> You could try to update your Gradle version to `5.0-20180903055829+0000` and then run it again.
> It should fix the problem (but raise other problems because you project is maybe not ready for Gradle 5.0 😅).
> 
> ```
> $ ./gradlew wrapper --gradle-version 5.0-20180903055829+0000 
> ```
> 
> As long as the fix is not included you have to move your code into a package..

i use android studio arctic fox stable version 

i still get the same error  ----    "not in a package"

**Comment by CMingTseng** [comment]
> `gradleVersion = '4.10'`
> 
> root project build.gradle (not kotlin file)
> 
> ```
> buildscript {
>     dependencies {
>         classpath XCfg.BuildPlugins.android_plugin
>     }
> }
> ```
> 
> buildSrc\build.gradle.kts
> 
> ```
> repositories {
>     jcenter()
> }
> plugins {
>     `kotlin-dsl`
> }
> ```
> 
> buildSrc\src\main\java\XCfg.kt
> 
> ```
> object XCfg {
>     object Versions {
>         val kotlinVersion = "1.2.61"
>         val gradleBuildTools = "3.3.0-alpha03"
>     }
> 
>     object BuildPlugins {
>         val android_plugin = "com.android.tools.build:gradle:${Versions.gradleBuildTools}"
>         val kotlin_plugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:${Versions.kotlinVersion}"
>     }
> }
> ```
> 
> ### Expected Behavior
> Should compile without any error
> 
> ### Current Behavior
> Failed on: ` No such property: BuildPlugins for class: XCfg`
> 
> ### Steps to Reproduce (for bugs)
> See file configuration on top of this issue.
> 
> When just change files to this, it start compiling and working.
> 
> root project build.gradle (not kotlin file)
> 
> ```
> buildscript {
>     dependencies {
>         classpath BuildPlugins.android_plugin
>     }
> }
> ```
> 
> buildSrc\src\main\java\XCfg.kt
> 
> ```
> object BuildPlugins {
>     val android_plugin = "com.android.tools.build:gradle:${Versions.gradleBuildTools}"
>     val kotlin_plugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:${Versions.kotlinVersion}"
> }
> 
> object Versions {
>     val kotlinVersion = "1.2.61"
>     val gradleBuildTools = "3.3.0-alpha03"
> }
> 
> object XCfg {
> }
> ```
> 
> But I think, that inner object should work for kotlin same as for groovy gradle files.
> In https://expertise.jetruby.com/how-to-write-gradle-scripts-with-kotlin-dsl-better-than-groovy-a63ffc9be2e3 they use this config, but for gradle kotlin files, not groovy. But I don't want to migrate all groovy files to kotlin files in same time for now...
@mtrakal  do you solve ? 

Now i  use  android studio arctic fox stable version  

gradle version   7.0.2  &  AGP 7.0.1 still get error

**Comment by mslalith** [comment]
Add this to `buildSrc/build.gradle.kts`
```
kotlin {
    // Add Deps to compilation, so it will become available in main project
    sourceSets.getByName("main").kotlin.srcDir("src/main/kotlin")
}
```

**Comment by EpicDima** [comment]
Since there is still no solution :( I have found several possible solutions for myself. Can be seen in the repository by commits
https://github.com/EpicDima/GradleBuildSrcIssue
